crates/astria-core/src/protocol/asset/v1alpha1/mod.rs

```diff
@@ -60,3 +64,186 @@ impl raw::DenomResponse {
         self.clone().into_native()
     }
 }
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct AllowedFeeAssetIdsResponse {
+    pub height: u64,
+    pub fee_asset_ids: Vec<asset::Id>,
+}
+
+impl AllowedFeeAssetIdsResponse {
+    /// Converts a protobuf [`raw::AllowedFeeAssetIdsResponse`] to an astria
+    /// native [`AllowedFeeAssetIdsResponse`].
+    ///
+    /// # Errors
+    /// - If one of the serialized asset IDs cannot be converted to a [`asset::Id`].
+    pub fn try_from_raw(
+        proto: &raw::AllowedFeeAssetIdsResponse,
+    ) -> Result<Self, IncorrectAssetIdLength> {
+        let raw::AllowedFeeAssetIdsResponse {
+            height,
+            fee_asset_ids,
+        } = proto;
+        let mut assets: Vec<asset::Id> = Vec::new();
+
+        for raw_id in fee_asset_ids {
+            // TODO: should this map_err to a `AllowedFeeAssetIdsResponseError`?
```

Comment:
i think it's ok to leave since it's the only error + unlikely to change this type